src/KilnMom.sol

```diff
@@ -0,0 +1,84 @@
+// SPDX-FileCopyrightText: © 2022 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.14;
+
+interface RugLike {
+    function rug(address dst) external;
+}
+
+interface AuthorityLike {
+    function canCall(address src, address dst, bytes4 sig) external view returns (bool);
+}
+
+// Bypass governance delay to disable a kiln instance
+contract KilnMom {
+    address public owner;
+    address public authority;
+
+    address public immutable dst;
+
+    event SetOwner(address indexed newOwner);
+    event SetAuthority(address indexed newAuthority);
+    event Rug(address indexed who, address indexed dst);
+
+    modifier onlyOwner {
+        require(msg.sender == owner, "KilnMom/only-owner");
+        _;
+    }
+
+    modifier auth {
+        require(isAuthorized(msg.sender, msg.sig), "KilnMom/not-authorized");
+        _;
+    }
+
+    constructor(address _dst) {
+        dst = _dst;
+        
+        owner = msg.sender;
+        emit SetOwner(msg.sender);
+    }
+
+    function isAuthorized(address src, bytes4 sig) internal view returns (bool) {
+        if (src == address(this)) {
+            return true;
+        } else if (src == owner) {
+            return true;
+        } else if (authority == address(0)) {
+            return false;
+        } else {
+            return AuthorityLike(authority).canCall(src, address(this), sig);
+        }
+    }
+
+    // Governance actions with delay
+    function setOwner(address owner_) external onlyOwner {
+        owner = owner_;
+        emit SetOwner(owner_);
+    }
+
+    function setAuthority(address authority_) external onlyOwner {
+        authority = authority_;
+        emit SetAuthority(authority_);
+    }
+
+    // Governance action without delay
+    function rug(address who) external auth {
+        address _dst = dst;
+        RugLike(who).rug(_dst);
```

Comment:
This is actually a change I requested (to make `dst` immutable and not define it as param).
`dst` will be the pauseProxy or maybe the `blow` in case of DAI (to send it back to the surplus buffer without selling it).
The problem of making `dst` a param in mom's `rug` is you will be allowing a malicious governance to get the funds without delay. I find that much more risky thing than realizing there is a problem in the recipient.

Reply:
hmm, i see your point, but if we are setting it to the `pauseProxy` anyways, then I'm not sure making it immutable really solves for that except in the cause of DAI going to `blow`.  I don't feel strongly if we feel the immutability is better

Reply:
Not sure I understand the last comment that if `dst` is set to `pauseProxy` then making it immutable (instead of a parameter) does not restrict the `chief`/`authority` from changing `dst` without a time delay.